internal/statefulsyncer/stateful_syncer.go

```diff
@@ -155,3 +157,60 @@ func (s *StatefulSyncer) BlockRemoved(
 
 	return err
 }
+
+// EndAtTipLoop runs a loop that evaluates end condition EndAtTip
+func (s *StatefulSyncer) EndAtTipLoop(
```

Comment:
@slowboat0 How do you feel about putting all end condition checks in a single go routine?

Reply:
When I initially thought about this PR, I thought about doing it that way. However, there is a cleanliness in using separate goroutines.

Reply:
I'd like to keep it this way. One caveat of using single goroutine is one expensive check on the condition (http timeout, db query timeout..) can block other condition checking as well.

I'm open to refactor once we've added more end conditions.